File: app/src/main/java/com/github/adrijanrogan/etiketa/ui/StartActivity.java
package com.github.adrijanrogan.etiketa.ui;

import android.Manifest;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;
import androidx.appcompat.app.AppCompatActivity;
import android.widget.Toast;

import com.github.adrijanrogan.etiketa.R;


public class StartActivity extends AppCompatActivity {

    private static final int PERMISSION_REQUEST = 9858;

    // Preverimo, ali imamo dovoljenje za dostop do spomina.
    // Ce dovoljenja nimamo, ga zahtevamo.
    // Ce dovoljenje imamo, zazenemo BrowserActivity
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_start);

        int permissionCheck = ContextCompat.checkSelfPermission
                (this, Manifest.permission.WRITE_EXTERNAL_STORAGE);
        if (permissionCheck == PackageManager.PERMISSION_GRANTED) {
            proceed();
        } else {
            requestPermission();
        }
    }

    // Zahtevamo dovoljenje za dostop do spomina
    private void requestPermission() {
        ActivityCompat.requestPermissions(this,
                new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE}, PERMISSION_REQUEST);
    }

    // Povratni klic, ki ga dobimo, ko se uporabnik odloci, ali bo zahtevo sprejel ali zavrnil
    // Ce zahtevo sprejme, zazenemo BrowserActivity, sicer pa se enkrat zahtevamo pravico
    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        if (requestCode == PERMISSION_REQUEST) {
            if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                proceed();
            } else {
                Toast.makeText(this,
                        "Prosimo sprejmite zahtevo za dostop do pomnilnika",
                        Toast.LENGTH_LONG).show();
            }
        }
    }

    // Zazenemo BrowserActivity
    private void proceed() {
        Intent intent = new Intent(this, BrowserActivity.class);
        startActivity(intent);
    }
}

File: app/src/main/java/com/github/adrijanrogan/etiketa/ui/StartActivity.kt
package com.github.adrijanrogan.etiketa.ui

import android.Manifest
import android.content.Intent
import android.content.pm.PackageManager
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import com.github.adrijanrogan.etiketa.R
import com.github.adrijanrogan.etiketa.ui.browse.BrowseActivity

class StartActivity : AppCompatActivity() {

    // Nimamo dovoljenja -> zahtevamo
    // sicer -> zazenemo BrowseActivity
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_start)

        val permissionCheck = ContextCompat.checkSelfPermission(
                this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
        if (permissionCheck == PackageManager.PERMISSION_GRANTED) {
            proceed()
        } else {
            requestPermission()
        }
    }

    private fun requestPermission() {
        ActivityCompat.requestPermissions(this,
                arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE), PERMISSION_REQUEST)
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>,
                                            grantResults: IntArray) {
        if (requestCode == PERMISSION_REQUEST) {
            if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                proceed()
            } else {
                Toast.makeText(this,
                        "Prosimo sprejmite zahtevo za dostop do pomnilnika",
                        Toast.LENGTH_LONG).show()
            }
        }
    }

    private fun proceed() {
        val intent = Intent(this, BrowseActivity::class.java)
        startActivity(intent)
    }

    companion object {
        private const val PERMISSION_REQUEST = 9858
    }
}